apps/chat2/chat2.nim

```diff
@@ -545,8 +545,8 @@ proc processInput(rfd: AsyncFD) {.async.} =
           echo "You are registered to the rln membership contract, find details of your registration transaction in https://goerli.etherscan.io/tx/0x", txHash
         
         echo "rln-relay preparation is in progress..."
-        let res = node.mountRlnRelay(conf = conf, spamHandler = some(spamHandler), registrationHandler = some(registrationHandler))
-        if res.isErr:
+        let res = await node.mountRlnRelay(conf = conf, spamHandler = some(spamHandler), registrationHandler = some(registrationHandler))
```

Comment:
Why treat it as an async proc?

Reply:
Because the dynamic mounting proc is async. We're trying to avoid use of waitFor unless absolutely necessary (i.e in entrypoints)

Reply:
> Because the dynamic mounting proc is async. We're trying to avoid use of waitFor unless absolutely necessary (i.e in entrypoints)

Exactly, this is the rule we are following. Only use `waitFor` at the application level (e.g., `chat2` or `wakunode2`)

The reason is that using `waitFor` indiscriminately can lead to a live-lock situation where the chronos' futures won't progress. This situation can happen if we wait for a method that (at some point deep in the call hierarchy) also contains a `waitFor`. The clearest example is `asyncTest` method: which contains a `waitFor` keeping the chronos' scheduler polling until all the asynchronous methods/futures have been completed.